hilo.rb

```diff
@@ -8,17 +8,38 @@ def initialize
   end
 
   def play
+    previous_guesses = []
+
     while @guesses.positive?
       puts 'Guess the number between 1 & 100 (inclusive):'
-      guess = gets.chomp.to_i
+      guess = nil
+
+      loop do
+        input = gets.chomp
+        if input.match?(/^\d+$/)
+          guess = input.to_i
+          if guess.between?(1, 100)
+            if previous_guesses.include?(guess)
+              puts 'You have already guessed that number, try a different number:'
+            else
+              previous_guesses << guess
+              break
+            end
+          else
+            puts 'Invalid input, please enter a number between 1 & 100:'
+          end
+        else
+          puts 'Invalid input, please enter a number between 1 & 100:'
+        end
```

Comment:
It would be nice to extract some of this logic into a validate guess method or something like that.